### ISSUE TITLE
yolov8 onnx模型静态量化

### ISSUE THREAD

**Issue by venxzw** [ultralytics/ultralytics#16439]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

使用onnxruntime自带的静态量化函数量化训练得到的yolov8模型后预测会在每处多出一个box，有大佬知道原因吗？量化代码如下：
`import numpy as npfrom onnxruntime.quantization import CalibrationDataReader, quantize_static, QuantType, QuantFormat    # Class for Callibration Data readingclass ImageCalibrationDataReader(CalibrationDataReader):    def __init__(self, image_paths):        self.image_paths = image_paths        self.idx = 0        self.input_name = "images"
    def preprocess(self, frame):        # Same preprocessing that you do before feeding it to the model        frame = cv2.imread(frame)        X = cv2.resize(frame, (640, 640))        image_data = np.array(X).astype(np.float32) / 255.0  # Normalize to [0, 1] range        image_data = np.transpose(image_data, (2, 0, 1))  # (H, W, C) -> (C, H, W)        image_data = np.expand_dims(image_data, axis=0)  # Add batch dimension        return image_data
    def get_next(self):        # method to iterate through the data set        if self.idx >= len(self.image_paths):            return None
        image_path = self.image_paths[self.idx]        input_data = self.preprocess(image_path)        self.idx += 1        return {self.input_name: input_data}
# Assuming you have a list of image paths for calibrationcalibration_image_paths = ['test.jpg'] # you can add more of the image paths
# Create an instance of the ImageCalibrationDataReadercalibration_data_reader = ImageCalibrationDataReader(calibration_image_paths)
# Use the calibration_data_reader with quantize_staticquantize_static('preprocessed.onnx', "static_quantized.onnx",                weight_type=QuantType.QInt8,                activation_type=QuantType.QUInt8,                calibration_data_reader=calibration_data_reader,                quant_format=QuantFormat.QDQ,                nodes_to_exclude=['/model.22/Concat_3', '/model.22/Split', '/model.22/Sigmoid'                                 '/model.22/dfl/Reshape', '/model.22/dfl/Transpose', '/model.22/dfl/Softmax',                                  '/model.22/dfl/conv/Conv', '/model.22/dfl/Reshape_1', '/model.22/Slice_1',                                 '/model.22/Slice', '/model.22/Add_1', '/model.22/Sub', '/model.22/Div_1',                                  '/model.22/Concat_4', '/model.22/Mul_2', '/model.22/Concat_5'],                per_channel=False,                reduce_range=True,)`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @venxzw, thank you for your interest in Ultralytics 🚀! We recommend checking out our [Docs](https://docs.ultralytics.com) first, which can be particularly helpful for new users. You may find relevant [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples and solutions to common issues.

It looks like you're encountering a challenge with model quantization resulting in extra boxes per prediction. If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/)? This will help us understand and address the issue better.

For custom training ❓ questions, sharing detailed information such as dataset samples and training logs is crucial. Ensure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the community discussion where it suits you best:
- [Discord](https://ultralytics.com/discord) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com) for in-depth discussions
- [Subreddit](https://reddit.com/r/ultralytics) for community knowledge sharing

## Upgrade

Make sure you're using the latest version of the `ultralytics` package. Update with:

```bash
pip install -U ultralytics
```

Ensure all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are met in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

You can run YOLOv8 in the following environments with dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn):

- **Notebooks** with a GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

For complete environment setups, refer to the guides for [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/), and [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This automated response is here to help you get started, but an Ultralytics engineer will reach out soon to assist you further.

**Comment by glenn-jocher** [comment]
@venxzw it seems like the issue might be related to the nodes you're excluding during quantization. Try reviewing the nodes specified in `nodes_to_exclude` to ensure they are correctly identified and necessary for exclusion. Adjusting these exclusions might help resolve the extra bounding box predictions.
